mcp/streamable_test.go

```diff
@@ -211,8 +211,40 @@ func TestClientReplay(t *testing.T) {
 	}
 }
 
-// Helper to read a specific number of progress notifications.
-func readProgressNotifications(t *testing.T, ctx context.Context, notifications chan string, count int) []string {
+// TestServerInitiatedSSE verifies that the persistent SSE connection remains
+// open and can receive server-initiated events.
+func TestServerInitiatedSSE(t *testing.T) {
+	notifications := make(chan string)
+	server := NewServer(testImpl, nil)
+
+	httpServer := httptest.NewServer(NewStreamableHTTPHandler(func(*http.Request) *Server { return server }, nil))
+	defer httpServer.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	client := NewClient(testImpl, &ClientOptions{ToolListChangedHandler: func(ctx context.Context, cc *ClientSession, params *ToolListChangedParams) {
+		notifications <- "toolListChanged"
+	},
+	})
+	clientSession, err := client.Connect(ctx, NewStreamableClientTransport(httpServer.URL, nil))
+	if err != nil {
+		t.Fatalf("client.Connect() failed: %v", err)
+	}
+	defer clientSession.Close()
+	time.Sleep(50 * time.Millisecond)
```

Comment:
I don't believe you need this. (If you do, we have a race condition.) 
The server side will enqueue all messages for stream 0 until the GET occurs.